legend-engine-xt-serviceStore-pure/src/main/resources/core_servicestore/executionPlan/executionPlan_generation.pure

```diff
@@ -220,14 +221,15 @@ function meta::external::store::service::executionPlan::generation::nodeFromServ
                                                                resultKey = $serviceBodyResolutionNode->toOne()->cast(@AllocationExecutionNode).varName));
 
   let mappedParameters           = $serviceMapping.requestBuildInfo.requestParametersBuildInfo.parameterBuildInfoList.serviceParameter;
-  let serviceStoreNode           = ^RestServiceExecutionNode(url             = $s.baseUrl + $serviceMapping.service.resolveFullPathRecursively(),
-                                                             method          = $serviceMapping.service.method,
-                                                             mimeType        = $serviceMapping.service.response.binding.contentType,
-                                                             params          = $serviceMapping.service.parameters,
-                                                             securitySchemes = $serviceMapping.service.security,
+  let serviceStoreNode           = ^RestServiceExecutionNode(url                    = $s.baseUrl + $serviceMapping.service.resolveFullPathRecursively(),
+                                                             method                 = $serviceMapping.service.method,
+                                                             mimeType               = $serviceMapping.service.response.binding.contentType,
+                                                             params                 = $serviceMapping.service.parameters,
+                                                             securitySchemes        = $serviceMapping.service.security->map(a | $a->cast(@SingleSecuritySchemeRequirement).securityScheme),
+                                                             authSchemes            = $serviceMapping.service.security->map(a | ^SingleAuthScheme(securityScheme = $a->cast(@SingleSecuritySchemeRequirement).securityScheme, authenticationSpecification = $s.authSpecs->get($a->cast(@SingleSecuritySchemeRequirement).id)->toOne())),
```

Comment:
Lets add a comment why we make this assumption that we will always have SingleSecuritySchemeRequirement